lib/matplotlib/backends/qt_compat.py

```diff
@@ -112,12 +118,20 @@ def _isdeleted(obj):
 if QT_API in [QT_API_PYQT6, QT_API_PYQT5, QT_API_PYSIDE6, QT_API_PYSIDE2]:
     _setup_pyqt5plus()
 elif QT_API is None:  # See above re: dict.__getitem__.
-    _candidates = [
-        (_setup_pyqt5plus, QT_API_PYQT6),
-        (_setup_pyqt5plus, QT_API_PYSIDE6),
-        (_setup_pyqt5plus, QT_API_PYQT5),
-        (_setup_pyqt5plus, QT_API_PYSIDE2),
-    ]
+    if _QT_MODE is None:
+        _candidates = [
+            (_setup_pyqt5plus, QT_API_PYQT6),
+            (_setup_pyqt5plus, QT_API_PYSIDE6),
+            (_setup_pyqt5plus, QT_API_PYQT5),
+            (_setup_pyqt5plus, QT_API_PYSIDE2),
+        ]
+    elif _QT_MODE == 5:
```

Comment:
Should we not have a `_QT_MODE == 6` case? If you explicitly import `backend_qt6`, then you would expect Qt6 and not a fallback to Qt5.

Reply:
When going to implement this, we do not _have_ a `backend_qt6`.  Because @anntzer shimmed them together well enough all of the selection logic is now in `qt_compat`. 

However, the only reliably way to control with qt major version we use first is to import it first.  We probably should do something about that...

Reply:
My intent was to ultimately get rid of backend_qt5foo (well, in the sense of letting it die its natural death when qt5 itself dies) in favor of a single backend_qtfoo and QT_API/preimports as selection method (the reason being that backend_qt5foo is enough to select between qt5 and qt6, but still not enough to select between pyqt and pyside, so it's a complete selection method to start with).  But perhaps deprecating the backend_qt5foo module would actually be less confusing for the end users... (still allowing `use("qt5foo")` because of its widespread use and automapping that to `use("qtfoo")`).

Another thing we may consider doing and which would have helped with https://github.com/matplotlib/matplotlib/issues/21998 is that the first time we instantiate a FigureCanvasQT, we check for all already imported qt bindings (by going through sys.modules) and, for any of them already imported, check if QApplication.instance() is not None; if the QApplication's binding does not match the one selected by qt_compat (which may occur e.g. because the qt binding was explicitly imported *after* pyplot as in #21998), then error out with a suitable message.

Reply:
I agree with the goal of letting the qt-version backends drop off when qt5 does (and plan to update the shims for Qt7).

Agree about adding the warnings when creating the QApplication.

I had discounted the power of `QT_API`, but think we should also add some warnings if we drop that on the floor because it gets pre-empted pre-existing imports.   Do we also want to try and detect if the user has pre-imported more than one and warn / explode?

Reply:
Actually, it looks like we should *not* prevent users from importing incompatible bindings, and even the check I suggested above may be too much; see https://github.com/spyder-ide/qtpy/issues/192.  (Let's not think about adding FORCE_QT_API kinds of things.)

Reply:
🤯 Ok.  I see why that works, it is just a bit unsettling that it does.   If you are using pyplot, then we do not actually need to parent any of our widgets into the users so if you were to have a Qt app that used pyplot to spawn windows everything should work.

If we do not want to check pyqt vs pyside, do we want to at least check QT5 vs QT6?

Agree going down the "force" route is not a good idea.  We are a library so the user _always_ has the ability to import what ever they want ahead of us.

Reply:
Checking qt5/qt6 seems reasonable(?).

Reply:
This will now warn but sometimes segfault the tests, going to say this is resolved.